operator/v1/types_csi_cluster_driver.go

```diff
@@ -109,23 +109,25 @@ type ClusterCSIDriverSpec struct {
 }
 
 // CSIDriverType indicates type of CSI driver being configured.
-// +kubebuilder:validation:Enum="";AWS;Azure;GCP;vSphere
+// +kubebuilder:validation:Enum="";AWS;Azure;GCP;IBMCloud;vSphere
 type CSIDriverType string
 
 const (
-	AWSDriverType     CSIDriverType = "AWS"
-	AzureDriverType   CSIDriverType = "Azure"
-	GCPDriverType     CSIDriverType = "GCP"
-	VSphereDriverType CSIDriverType = "vSphere"
+	AWSDriverType      CSIDriverType = "AWS"
+	AzureDriverType    CSIDriverType = "Azure"
+	GCPDriverType      CSIDriverType = "GCP"
+	IBMCloudDriverType CSIDriverType = "IBMCloud"
+	VSphereDriverType  CSIDriverType = "vSphere"
 )
 
 // CSIDriverConfigSpec defines configuration spec that can be
 // used to optionally configure a specific CSI Driver.
+// +kubebuilder:validation:XValidation:rule="has(self.driverType) && self.driverType == 'IBMCloud' ? (has(self.ibmcloud) && size(self.ibmcloud.encryptionKeyCRN) > 0) : !has(self.ibmcloud)",message="ibmcloud must be set if driverType is 'IBMCloud', but remain unset otherwise"
```

Comment:
Why `encryptionKeyCRN > 0` in this? That implies to me that the encryption key should be a required field?

Reply:
If we make it required now, we could make it optional in the future when we add a second field to the mix

Reply:
I changed `ecnryptionKey` to required

Reply:
You should remove the encryption key part from this CEL validation
```suggestion
// +kubebuilder:validation:XValidation:rule="has(self.driverType) && self.driverType == 'IBMCloud' ? has(self.ibmcloud) : !has(self.ibmcloud)",message="ibmcloud must be set if driverType is 'IBMCloud', but remain unset otherwise"
```